dynamodb-stream-reader/src/test/java/no/unit/nva/dynamodb/DynamoDBStreamHandlerTest.java

```diff
@@ -0,0 +1,128 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+
+@SuppressWarnings("unchecked")
+public class DynamoDBStreamHandlerTest {
+
+    private static final String SAMPLE_MODIFY_EVENT_FILENAME = "DynamoDBStreamModifyEvent.json";
+    private static final String SAMPLE_INSERT_EVENT_FILENAME = "DynamoDBStreamInsertEvent.json";
+    private static final String SAMPLE_REMOVE_EVENT_FILENAME = "DynamoDBStreamRemoveEvent.json";
+    private static final String SAMPLE_UNKNOWN_EVENT_FILENAME = "UnknownDynamoDBEvent.json";
+    public static final String ELASTICSEARCH_ENDPOINT_ADDRESS = "localhost";
+    private static final String ELASTICSEARCH_ENDPOINT_INDEX = "resources";
+    private static final String ELASTICSEARCH_ENDPOINT_API_SCHEME = "http";
+    private static final Object TARGET_SERVICE_URL = "http://localhost/service/";
+
+    private DynamoDBStreamHandler handler;
+    private ElasticSearchRestClient elasticSearchRestClient;
+    private Environment environment;
+    private Context context;
+    private HttpClient httpClient;
+
+    /**
+     * Set up test environment.
+     *
+     * @throws IOException some error occurred
+     * @throws InterruptedException another error occurred
+     */
+    @BeforeEach
+    public void init() throws IOException, InterruptedException {
+        environment = spy(Environment.class);
+        context = mock(Context.class);
+        httpClient = spy(HttpClient.class);
+        doReturn(ELASTICSEARCH_ENDPOINT_ADDRESS).when(environment)
+                .readEnv(ElasticSearchRestClient.ELASTICSEARCH_ENDPOINT_ADDRESS_KEY);
+        doReturn(ELASTICSEARCH_ENDPOINT_INDEX).when(environment)
+                .readEnv(DynamoDBStreamHandler.ELASTICSEARCH_ENDPOINT_INDEX_KEY);
+        doReturn(ELASTICSEARCH_ENDPOINT_API_SCHEME).when(environment)
+                .readEnv(ElasticSearchRestClient.ELASTICSEARCH_ENDPOINT_API_SCHEME_KEY);
+
+        doReturn(TARGET_SERVICE_URL).when(environment)
+                .readEnv(DynamoDBStreamHandler.TARGET_SERVICE_URL_KEY);
+        doReturn(ELASTICSEARCH_ENDPOINT_INDEX).when(environment)
+                .readEnv(DynamoDBStreamHandler.ELASTICSEARCH_ENDPOINT_INDEX_KEY);
+
+        HttpResponse<String> successResponse = mock(HttpResponse.class);
+        doReturn(successResponse).when(httpClient).send(any(), any());
+        elasticSearchRestClient = new ElasticSearchRestClient(httpClient, environment);
+        handler = new DynamoDBStreamHandler(elasticSearchRestClient, environment);
+    }
+
+    @Test
+    @DisplayName("testCreateHandlerWithEmptyEnvironmentShouldFail")
+    public void testCreateHandlerWithEmptyEnvironmentShouldFail()  {
+        assertThrows(IllegalStateException.class, () -> new DynamoDBStreamHandler());
```

Comment:
test the message. assertThrows returns the thrown exception so you can test the message as well.

Reply:
Testing that there IS an environment variabe missing (not which one)